src/services/Subscriptions.php

```diff
@@ -99,8 +102,18 @@ public function createOrUpdateSubscription(StripeSubscription $subscription): bo
             ->one();
 
         if ($subscriptionElement === null) {
-            /** @var SubscriptionElement $subscriptionElement */
-            $subscriptionElement = new SubscriptionElement();
+            if ($checkForUnsavedDraft) {
```

Comment:
Is this param needed? Could we not just check if the retrieved element is a draft and then do the logic later in this method based on that?

Reply:
To get the unsaved draft, we need to query Stripe to get the session by subscription id. From that we need to extract the `craftSubscriptionUid` metadata and use that to query for the unsaved draft (by uid and including drafts). 
So, for me, it makes more sense to have that param and do all that work only when needed, which should be only in response to receiving a `customer.subscription.created` webhook. 
I’m happy to discuss this further, though!

Reply:
It’s now updated as per our chat.